### PR TITLE
Interface for UPPAALDriver and Singleton for management

### PR DESCRIPTION
**Pull request by Nielswps**

### CODE REVIEW

File: src/main/java/dk/cs/aau/huppaal/controllers/HUPPAALController.java [hunk 1]
```diff
@@ -403,6 +387,27 @@ private void initializeMenuBar() {
             HUPPAAL.save();
         });
 
+        menuBarPreferencesUppaalLocation.setOnAction(event -> {
+            // Dialog title
+            final FileChooser filePicker = new FileChooser();
+            filePicker.setTitle("Choose UPPAAL server file");
+            filePicker.getExtensionFilters().add(new FileChooser.ExtensionFilter("UPPAAL server", "*.exe"));
```
Comment (yrke) [inline src/main/java/dk/cs/aau/huppaal/controllers/HUPPAALController.java]: on macOs/linux the server will not include *.exe

Comment (Nielswps) [reply src/main/java/dk/cs/aau/huppaal/controllers/HUPPAALController.java]: I have removed the filter, I'm unsure whether the file type should be handled in any way or just caught by exceptions when trying to run it

**Review by yrke — CHANGES_REQUESTED** [review]
When uppaal server file it located, check the server and clear the warning about uppaal not found.

**Comment by yrke** [comment]
Also i get an "java.util.MissingResourceException: Can't find bundle for base name locale.LexicalAnalyser, locale en_US" in the background don't know if this is related. 
And also having some problems with the verifyer locking up, byt maybe we should look at this in an other PR.

**Comment by Nielswps** [comment]
I have removed the warning when the UPPAALDriver is set, but the interface does not correspond with this. I'm unsure how to force the interface to be updated

**Review by yrke — APPROVED** [review]

**Pull request merged**